### PR TITLE
Yoda configurable timeout duration

### LINKED ISSUE #2258: Yoda configurable timeout duration
Users can adjust the timeout for each executor for protecting executor source from a malicious data source. You need to design where and how we can config timeout for each executor.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #2258 

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

## If this PR doesn't need to be merged into v1.0.6 please remove this option:

- [ ] The pull request has been cherry-picked to `rc/1.0.6`

### CODE REVIEW

File: chain/yoda/executor/executor.go [hunk 1]
```diff
@@ -18,30 +20,47 @@ type ExecResult struct {
...
+	if err != nil {
+		return "", "", 0, fmt.Errorf("Invalid url, cannot parse url: %s", executor[1])
+	}
+
+	query := u.Query()
+	timeoutStr := query.Get(flagQueryTimeout)
+	if timeoutStr == "" {
+		timeoutStr = "10s"
+	}
+	query.Del(flagQueryTimeout)
```
Comment (taobun) [inline chain/yoda/executor/executor.go:62]: Add comment why we must do this

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/yoda/executor/executor.go [hunk 2]
```diff
@@ -18,30 +20,47 @@ type ExecResult struct {
...
+		return "", "", 0, fmt.Errorf("Invalid executor, cannot parse executor: %s", executorStr)
 	}
-	return executor[0], executor[1], nil
+	u, err := url.Parse(executor[1])
+	if err != nil {
+		return "", "", 0, fmt.Errorf("Invalid url, cannot parse url: %s", executor[1])
+	}
+
+	query := u.Query()
+	timeoutStr := query.Get(flagQueryTimeout)
```
Comment (sorawit) [inline chain/yoda/executor/executor.go:57]: Let's make this required.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/yoda/executor/docker_test.go [hunk 3]
```diff
@@ -6,8 +6,8 @@ import (
 
 func TestDockerSuccess(t *testing.T) {
 	// TODO: Enable test when CI has docker installed.
-	// e := NewDockerExec("bandprotocol/runtime:1.0.1")
-	// 	res, err := e.Exec(10*time.Second, []byte(`#!/usr/bin/env python3
+	// 	e := NewDockerExec("bandprotocol/runtime:1.0.1", 20*time.Second)
```
Comment (taobun) [inline chain/yoda/executor/docker_test.go]: 10 second and use runtime 1.0.2 version

File: chain/yoda/executor/docker_test.go [hunk 4]
```diff
@@ -36,13 +36,13 @@ func TestDockerSuccess(t *testing.T) {
...
 	// 	fmt.Println(string(res.Output), res.Code, err)
-	// require.True(t, false)
+	// 	require.True(t, false)
 }
 
 func TestDockerLongStdout(t *testing.T) {
 	// TODO: Enable test when CI has docker installed.
-	// 	e := NewDockerExec("bandprotocol/runtime:1.0.1")
-	// 	res, err := e.Exec(10*time.Second, []byte(`#!/usr/bin/env python3
+	// 	e := NewDockerExec("bandprotocol/runtime:1.0.1", 10*time.Second)
```
Comment (taobun) [inline chain/yoda/executor/docker_test.go]: ditto

File: chain/yoda/executor/executor.go [hunk 5]
```diff
@@ -3,11 +3,13 @@ package executor
 import (
 	"errors"
 	"fmt"
+	"net/url"
 	"strings"
 	"time"
 )
 
 var (
+	flagQueryTimeout   = "timeout"
```
Comment (taobun) [inline chain/yoda/executor/executor.go]: I think we can use 
```
const (
    flagQueryTimeout   = "timeout"
)
```

File: chain/yoda/executor/executor.go [hunk 6]
```diff
@@ -18,30 +20,48 @@ type ExecResult struct {
...
+		return "", "", 0, fmt.Errorf("Invalid executor, cannot parse executor: %s", executorStr)
 	}
-	return executor[0], executor[1], nil
+	u, err := url.Parse(executor[1])
+	if err != nil {
+		return "", "", 0, fmt.Errorf("Invalid url, cannot parse url: %s", executor[1])
+	}
+
+	query := u.Query()
+	timeoutStr := query.Get(flagQueryTimeout)
```
Comment (taobun) [inline chain/yoda/executor/executor.go:54]: timeoutParams, found := query[flagQueryTimeout]
...
timeout, err = time.ParseDuration(timeoutParams[0])

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**